internal/daemon/vendor-specific-plugins/marvell/main.go

```diff
@@ -88,25 +88,37 @@ func (vsp *mrvlVspServer) createVethPair(index int) error {
 	//dpInterfaceName is the name of the interface on the Data Plane side
 	secInterfaceName := fmt.Sprintf("nf_interface%d", index)
 	dpInterfaceName := fmt.Sprintf("dp_interface%d", index)
-	vethLink := &netlink.Veth{
-		LinkAttrs: netlink.LinkAttrs{Name: secInterfaceName},
-		PeerName:  dpInterfaceName,
-	}
-	if err := netlink.LinkAdd(vethLink); err != nil {
-		return err
+
+	var nfLink netlink.Link
+	var peerLink netlink.Link
+	var err error
+
+	nfLink, _ = netlink.LinkByName(secInterfaceName)
```

Comment:
Why don't we check err on LinkByName?

Isn't it possible this fails on the by LinkByName()call to "execGetLink()".  Might be easier for debugging to catch/report this error earlier?

Reply:
I think we strongly rely that the API guarantees `(nfLink != nil) == (err == nil)`?

If we would not rely on such conditions, we would always need to double check
```go
    if err != nil {
         return err
    }
    if nfLink == nil {
        return errors.New("no device returned")
    }
```
(which in general we don't do, because it's redundant and because we really want an error reason when the function fails to provide a result).

Note that below there is the check `if nfLink == nil || peerLink == nil {`, so we do check for error still. We just don't care about the actual `err` variable.

Maybe that's unclear and I should rework this to check for `err` instead. Is that what you suggest?

Reply:
I think it is fine, it just seems like we would lose any unexpected error message returned by netlink.LinkByName()

Reply:
Maybe I am misunderstanding as well...so then we expect that this function will fail some of the time, and if it fails for either the nfLink or Peer link, we know we should try to create the vethLink again? Is that correct?

Reply:
the function first tries to fetch `nfLink` and `peerLink`.

If that succeeds, they are both taken.

If that fails (`if nfLink == nil || peerLink == nil {`), we create the veth pair. This is a check for error, it's just we don't care about the actual `error` variable, because there is nothing to be done about that.

> it just seems like we would lose any unexpected error message returned by netlink.LinkByName().

Those errors are not used. They are not lost, they are intentionally ignored.  Note that we do check for error  (`if nfLink == nil || peerLink == nil {`), just not for the `error` variable.